adagrams/game.py

```diff
@@ -1,11 +1,121 @@
+import random
+
 def draw_letters():
-    pass
+    letter_pool = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
+
+    hand = []
+
+    # Randomly get letters from letter pool while decrementing the value associated with it
+    while len(hand) < 10:
+        letter = random.choice(list(letter_pool))
+        if letter_pool[letter] > 0:
+            hand.append(letter)
+            letter_pool[letter] -= 1
+            
+    return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # Create a copy of letter bank
+    letters_copy = letter_bank.copy()
+
+    # Loop through each letter of word and remove letter from letters copy
+    for letter in word:
+        if letter.upper() in letters_copy:
+            letters_copy.remove(letter.upper())
```

Comment:
I like your decision to work with `letter.upper()` in the conditional (rather than `word = word.upper()`)! I think it feels less like you're modifying some input.